lib/kino/js/live/server.ex

```diff
@@ -116,4 +142,28 @@ defmodule Kino.JS.Live.Server do
 
     :ok
   end
+
+  defp add_client(ctx, pid, origin) do
+    if Map.has_key?(ctx.__private__.clients, origin) do
+      ctx
+    else
+      monitor_ref = Kino.Bridge.monitor(pid)
+      put_in(ctx.__private__.clients[origin], {pid, monitor_ref})
+    end
+  end
+
+  defp remove_client(ctx, ref) do
+    origin =
+      Enum.find_value(ctx.__private__.clients, fn
```

Comment:
It would be best to not do a full traversal. Maybe we could store the origin (or origins) in `__private__.refs`?

Reply:
Ah, my initial thought was that it's fine to do it whenever someone disconnects, but now I've realised we do this on every `:DOWN` messages that the user may be expecting. I will optimise with a separate map.